### ISSUE TITLE
Style/HashSyntax with ruby 3.1 style produces either failing cop OR buggy code

### ISSUE THREAD

**Issue by dewyze** [rubocop/rubocop#11544]
## Expected behavior

```yml
Style/HashSyntax:
  EnforcedShorthandSyntax: consistent
```

I am writing a request spec:

```ruby
params = { user: { email: "user@example.com" } }

post "/users", params: params

<Any code here>
```

Rubocop will say that the `post` line is a violation, but I take the suggestion and remove the trailing `params`, then the ruby code will not compile (or throw an error at runtime).

I think Rubocop should not enforce the shorthand when it's at the end of a line if there are no closing brackets/parentheses. It leads to buggy ruby.

## Actual behavior

Rubocop turns `post "/users", params: params` into `post "/users", params:`.

However, without parentheses, rubocop will parse the result of `<Any code here>` as the value for `params`.

## Steps to reproduce the problem

```yml
Style/HashSyntax:
  EnforcedShorthandSyntax: consistent
```

```ruby
def post(endpoint, params:)
  puts "POST #{endpoint} with #{params}"
end

params = { a: 1 }

post "/users", params:

puts "Success!"
```

Rubocop will accept this, but ruby will throw an error.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
→ rubocop -V
1.44.1 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [arm64-darwin22]
  - rubocop-performance 1.14.2
  - rubocop-rails 2.15.2
  - rubocop-rspec 2.12.1
  - rubocop-sorbet 0.7.0
```

**Comment by sunny** [comment]
Related to #11408

**Comment by jcoyne** [comment]
Can this change be at least marked autocorrect unsafe?

**Comment by koic** [comment]
Parentheses are added to prevent the syntax incompatibility in the latest RuboCop.
